### PR TITLE
Feature: estimote android scanner && geolocation based beacon fetch

### PR DESCRIPTION
**Pull request by berkayinceisci**

### CODE REVIEW

**Review by baseru — APPROVED** [review]

File: kmmsdk/src/androidMain/kotlin/com/lokate/kmmsdk/AndroidEstimoteBeaconScanner.kt [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        val (beaconUUID, major, minor) = regionContext.tag.split(":")
+
+        scanResults.add(
+            BeaconScanResult(
+                beaconUUID = beaconUUID,
+                major = major.toInt(),
+                minor = minor.toInt(),
+                rssi = 0.0, // not relevant (handled inside EstimoteSDK)
+                txPower = 0, // not relevant (handled inside EstimoteSDK)
+                accuracy = 0.0, // definitely inside proximity range, so passes the if check in LokateSDK
```
Comment (baseru) [inline kmmsdk/src/androidMain/kotlin/com/lokate/kmmsdk/AndroidEstimoteBeaconScanner.kt:129]: It looks like we cant filter this on LokateSDK while using estimote. It is better to make estimote handle the distance thing

File: kmmsdk/src/androidMain/kotlin/com/lokate/kmmsdk/AndroidEstimoteBeaconScanner.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+        Log.d("AndroidEstimoteBeaconScanner", "Setting regions")
+
+        this.regions.clear()
+        this.regions.addAll(
+            regions
+                .map { beacon ->
+                    ProximityZoneBuilder()
+                        .forTag(beacon.proximityUUID.uppercase() + ':' + beacon.major + ':' + beacon.minor)
+                        .inCustomRange(Double.MAX_VALUE)
```
Comment (baseru) [inline kmmsdk/src/androidMain/kotlin/com/lokate/kmmsdk/AndroidEstimoteBeaconScanner.kt:95]: https://github.com/lokateapp/LokateMobileSDK/pull/7/files#r1538288366

We should pass the distance estimote sdk does not allow us to get the accuracy field

Comment (berkayinceisci) [reply kmmsdk/src/androidMain/kotlin/com/lokate/kmmsdk/AndroidEstimoteBeaconScanner.kt:95]: yes, I was intentionally passing the distance before but forgot that during rebase.

Comment (baseru) [reply kmmsdk/src/androidMain/kotlin/com/lokate/kmmsdk/AndroidEstimoteBeaconScanner.kt:95]: I can update it on the iOS PR if it is okay

**Pull request merged**